### PR TITLE
mod_proxy_cluster.c: Code tweaks and polishing, removal of two asserts

### PR DESCRIPTION
**Pull request by jajik**
There are some tweaks, I removed some unnecessary duplication, and removed once-used variables when it made sense. I also removed the two asserts that I think are unnecessary, but it would be better to double check it, I might miss something.

### CODE REVIEW

File: native/mod_proxy_cluster/mod_proxy_cluster.c [hunk 1]
```diff
@@ -186,9 +186,7 @@ static char *normalize_hostname(apr_pool_t *p, const char *hostname)
     return ret;
 }
 
-/**
- * Normalize the worker name
- */
```
Comment (jfclere) [inline native/mod_proxy_cluster/mod_proxy_cluster.c:191]: I agree the format should be one and we should avoid mixing:
/*
 * Comment
 */
and
/* Comment */

But I would prefer keeping the first one, except you have a good justification.

File: native/mod_proxy_cluster/mod_proxy_cluster.c [hunk 2]
```diff
@@ -3156,11 +3090,10 @@ static void upd_context_count(const char *id, int val, const server_rec *s)
     }
 }
 
-static apr_status_t decrement_busy_count(void *worker_)
+static apr_status_t decrement_busy_count(void *w)
```
[the remark this answers is not in the document]
Comment (jajik) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:3093]: Ah, ok, so it's a function from the trunk, I did not notice that. Either way, we can leave it like this, we'll erase it in the future anyway.

File: native/mod_proxy_cluster/mod_proxy_cluster.c (hunk 1 above)
Comment (jajik) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:191]: Looking at it again, I would suggest leaving this as is and creating a new issue to unify the styling for the whole code base (and not only mod_proxy_cluster.c). WDYT?

Comment (jfclere) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:191]: Yes please ;-)

Comment (jajik) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:191]: Ok, I opened #158

File: native/mod_proxy_cluster/mod_proxy_cluster.c [hunk 3]
```diff
@@ -1741,7 +1700,7 @@ static proxy_worker *internal_find_best_byrequests(const proxy_balancer *balance
                 qsort(workers, workers_length, sizeof(*workers), &proxy_worker_cmp);
             }
             /* Compute consistent int from session id */
-            for (i = 0; session_id[i] != 0; ++i) {
+            for (i = 0; session_id[i] != 0; i++) {
```
[the remark this answers is not in the document]
Comment (jfclere) [reply native/mod_proxy_cluster/mod_proxy_cluster.c:1707]: it is good forget the question...

**Review by jfclere — APPROVED** [review]

**Comment by jajik** [comment]
@jfclere Thank you, merged!

**Pull request merged**